dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/AbstractTrackerPersister.java

```diff
@@ -112,6 +112,8 @@ public TrackerTypeReport persist(EntityManager entityManager, TrackerBundle bund
           determineNotificationTriggers(bundle.getPreheat(), trackerDto);
 
       try {
+        V originalEntity = cloneEntityProperties(bundle.getPreheat(), trackerDto);
```

Comment:
I know, but it can't because the next step, that converts the dto into a hibernate entity, updates the preheat event with the payload values of occurredDate, scheduledDate and geometry that's why I make a copy before that happens.

Reply:
This is the general issue I have with this abstract persister 😅 you can also see this from the noop implementations of some of the abstract methods.

What is code truly needed by all entities? Can we rather go for composition than inheritance to get rid of this awkward abstract class that does not seem to fit all entities anyway.